### PR TITLE
osfv_cli integration

### PR DESCRIPTION
**Pull request by PLangowski**

### CODE REVIEW

File: keywords.robot [hunk 1]
```diff
@@ -605,8 +620,12 @@ Prepare To PiKVM Connection
...
     Open Connection And Log In
     ${platform}=    Get Current RTE Param    platform
-    ${pikvm_ip}=    Get Current RTE Param    pikvm_ip
-    Set Global Variable    ${PIKVM_IP}
+    IF    '${SNIPEIT}' == 'yes'
+        ${pikvm_ip}=    Snipeit Get PiKVM IP    ${RTE_IP}
+        Set Global Variable    ${PIKVM_IP}
+    END
+    # If snipeit is set to "no", fetch it from command line
+    Variable Should Exist    ${PIKVM_IP}
```
Comment (macpijan) [inline keywords.robot:628]: add `msg=...` Explaining this in the error message

File: variables.robot [hunk 2]
```diff
@@ -201,6 +201,13 @@ ${OS_UBUNTU}=               ubuntu
 # MinnowBoard Turbot
 &{RTE55}=                   ip=192.168.10.112
 ...                         platform=minnowboard-turbot
+...                         sonoff_ip=192.168.4.35    pikvm_ip=192.168.4.180
```
Comment (macpijan) [inline variables.robot:204]: This PR may need a rebase after cleanup

File: keywords.robot [hunk 3]
```diff
@@ -546,6 +547,21 @@ Prepare Test Suite
...
+Import Osfv Libraries
+    [Documentation]    Import osfv_cli libraries based on config and command
+    ...    line variables
+    IF    '${SNIPEIT}' == 'yes'
+        Import Library    osfv.rf.snipeit_robot
+        Import Library    osfv.rf.rte_robot.RobotRTE    ${RTE_IP}    True
+    ELSE IF    'sonoff' == '${POWER_CTRL}'
+        Variable Should Exist    ${SONOFF_IP}
+        Import Library    osfv.rf.rte_robot.RobotRTE    ${RTE_IP}    False
+        ...    ${SONOFF_IP}    ${CONFIG}
```
Comment (macpijan) [inline keywords.robot:559]: What is CONFIG? Can we document this here?

File: scripts/regression.sh
[the remark this answers is not in the document]
Comment (PLangowski) [reply scripts/regression.sh:1]: I amended the commit, so it no longer modifies that file.

File: keywords.robot (hunk 1 above)
Comment (PLangowski) [reply keywords.robot]: Done 271316b396eccc0c67738c27e883e9bbf6eeeb54

File: keywords.robot (hunk 3 above)
Comment (PLangowski) [reply keywords.robot:562]: 271316b396eccc0c67738c27e883e9bbf6eeeb54

File: variables.robot (hunk 2 above)
Comment (PLangowski) [reply variables.robot]: I must have missed this while rebasing. I removed the line 3464d520c5ad486c12f082facf9331d400173e64

**Review by macpijan — COMMENTED** [review]
@PLangowski Code looks OK, but we still need to:
- [x] cleanup commits - we specifically do not want to merge commits like `wip`, or `refactor, if possible; we should split the whole changes into a few logical commits
- [ ] CI fails - you can reproduce it locally, it seems we need to adjust the command to start self-tests in QEMU
- [ ] Since we introduced some new variables, we should consider if we need to update other files where tests are executed; one example is the QEMU CI script, other will be README.md with documentation, or the `scripts/regression.sh` for performing regression

**Comment by PLangowski** [comment]
@macpijan I've done everything from the checklist.  The failing CIs were fixed by not importing the osfv_cli module when QEMU is used. This PR now has 5 commits and regression tests support the changes,

**Pull request closed without merging**